contracts/examples/DXswapSwapRelayer.sol

```diff
@@ -0,0 +1,296 @@
+        uint256 deadline;
+        uint256 maxWindowTime;
+        uint256 oracleId;
+        address factory;
+        bool executed;
+    }
+
+    uint256 public immutable GAS_ORACLE_UPDATE = 168364;
+    uint256 public immutable PARTS_PER_MILLION = 1000000;
+    uint256 public immutable BOUNTY = 0.01 ether; // To be decided
+    uint8 public immutable PROVISION = 1;
+    uint8 public immutable REMOVAL = 2;
+
+    address public immutable dxSwapFactory;
+    address public immutable dxSwapRouter;
+    address public immutable uniswapFactory;
+    address public immutable uniswapRouter;
+    address public immutable WETH;
+
+    OracleCreator oracleCreator;
+    uint256 public orderCount;
+    mapping(uint256 => Order) orders;
+    address payable public owner;
+
+    constructor(
+        address payable _owner,
+        address _dxSwapFactory,
+        address _dxSwapRouter,
+        address _uniswapFactory,
+        address _uniswapRouter,
+        address _WETH,
+        OracleCreator _oracleCreator
+    ) public {
+        owner = _owner;
+        dxSwapFactory = _dxSwapFactory;
+        dxSwapRouter = _dxSwapRouter;
+        uniswapFactory = _uniswapFactory;
+        uniswapRouter = _uniswapRouter;
+        WETH = _WETH;
+        oracleCreator = _oracleCreator;
+    }
+
+    function createSwapOrder(
+        address tokenA,
+        address tokenB,
+        uint256 amountInTokenA,
+        uint256 amountInTokenB,
+        uint256 priceTolerance,
+        uint256 minReserveA,
+        uint256 minReserveB,
+        uint256 maxWindowTime,
+        uint256 deadline,
+        address factory
+    ) external payable returns (uint256 orderIndex) {
+        require(factory == dxSwapFactory || factory == uniswapFactory, 'DXswapRelayer: INVALID_FACTORY');
+        require(msg.sender == owner, 'DXswapRelayer: CALLER_NOT_OWNER');
+        require(tokenA != tokenB, 'DXswapRelayer: INVALID_PAIR');
+        require(tokenA < tokenB, 'DXswapRelayer: INVALID_TOKEN_ORDER');
+        // only one token amount can be > 0 and second token amount is calculated as output
+        require(
+            (amountInTokenA > 0 && amountInTokenB == 0) || (amountInTokenA == 0 && amountInTokenB > 0),
+            'DXswapRelayer: INVALID_TOKEN_AMOUNT'
+        );
+        require(priceTolerance <= PARTS_PER_MILLION, 'DXswapRelayer: INVALID_TOLERANCE');
+        require(block.timestamp <= deadline, 'DXswapRelayer: DEADLINE_REACHED');
+        require(maxWindowTime > 30, 'DXswapRelayer: INVALID_WINDOWTIME');
+
+        if (tokenA == address(0)) {
+            require(address(this).balance >= amountInTokenA, 'DXswapRelayer: INSUFFICIENT_ETH');
+        } else {
+            require(IERC20(tokenA).balanceOf(address(this)) >= amountInTokenA, 'DXswapRelayer: INSUFFICIENT_TOKEN_A');
+        }
+        require(IERC20(tokenB).balanceOf(address(this)) >= amountInTokenB, 'DXswapRelayer: INSUFFICIENT_TOKEN_B');
+
+        address pair = _pair(tokenA, tokenB, factory);
```

Comment:
We could actually call the factory to get the pair address based on the 2 tokens that are into it. This will also let us know if the pair exists or not. If it doesn't we should abort here.